### PR TITLE
refactor: rework image build and repo structure

### PR DESCRIPTION
**Pull request by xoxys**

### CODE REVIEW

**Comment by xoxys** [comment]
Failed broken link check is expected.

**Comment by TheOneRing** [comment]
Sounds reasonable :D

File: .drone.star [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+  arches = [
+    'amd64',
+  ]
+
+  config = {
+    'version': None,
+    'arch': None,
+    'downstream': [],
+    'description': 'ownCloud Ubuntu base image',
```
Comment (wkloucek) [inline .drone.star:14]: Please change

File: .drone.star [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+    'settings': {
+      'username': {
+        'from_secret': 'internal_username',
+      },
+      'password': {
+        'from_secret': 'internal_password',
+      },
+      'tags': config['internal'],
+      'dockerfile': '%s/Dockerfile.%s' % (config['path'], config['arch']),
+      'repo': 'registry.drone.owncloud.com/owncloud/ubuntu',
```
Comment (wkloucek) [inline .drone.star:266]: Please change

File: .drone.star [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    'environment': {
+      'DOCKER_USER': {
+        'from_secret': 'internal_username',
+      },
+      'DOCKER_PASSWORD': {
+        'from_secret': 'internal_password',
+      },
+    },
+    'commands': [
+      'retry -- reg digest --username $DOCKER_USER --password $DOCKER_PASSWORD registry.drone.owncloud.com/owncloud/ubuntu:%s' % config['internal'],
```
Comment (wkloucek) [inline .drone.star:287]: needs to be changed?

File: .drone.star [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    'settings': {
+      'username': {
+        'from_secret': 'public_username',
+      },
+      'password': {
+        'from_secret': 'public_password',
+      },
+      'tags': config['tag'],
+      'dockerfile': '%s/Dockerfile.%s' % (config['path'], config['arch']),
+      'repo': 'owncloud/ubuntu',
```
Comment (wkloucek) [inline .drone.star:305]: doesn't look right

File: .drone.star [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+    'environment': {
+      'DOCKER_USER': {
+        'from_secret': 'internal_username',
+      },
+      'DOCKER_PASSWORD': {
+        'from_secret': 'internal_password',
+      },
+    },
+    'commands': [
+      'reg rm --username $DOCKER_USER --password $DOCKER_PASSWORD registry.drone.owncloud.com/owncloud/ubuntu:%s' % config['internal'],
```
Comment (wkloucek) [inline .drone.star:331]: needs a change?

**Review by wkloucek — CHANGES_REQUESTED** [review]

**Comment by xoxys** [comment]
@wkloucek Thanks! I've changed all reverences to use ctx/config to get the repo name.

**Comment by wkloucek** [comment]
@xoxys https://drone.owncloud.com/owncloud-docker/smashbox/110 the starlark file is broken

**Comment by xoxys** [comment]
Already fixed :)

**Review by wkloucek — APPROVED** [review]

**Pull request merged**